src/Domain/Structure/DirectionIdMap.hpp

```diff
@@ -0,0 +1,36 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <boost/functional/hash.hpp>
+#include <pup.h>
+
+#include "DataStructures/FixedHashMap.hpp"
+#include "Domain/Structure/DirectionId.hpp"
+#include "Domain/Structure/MaxNumberOfNeighbors.hpp"
+
+/// An optimized map with DirectionId keys
+template <size_t Dim, typename T>
+class DirectionIdMap
+    : public FixedHashMap<maximum_number_of_neighbors(Dim), DirectionId<Dim>, T,
+                          boost::hash<DirectionId<Dim>>> {
```

Comment:
Why not just leave the hasher defaulted? You added a `std::hash` for `DirectionId`